internal/services/compute/migration/legacy_vmss.go

```diff
@@ -808,6 +813,10 @@ func resourceArmVirtualMachineScaleSetExtensionHash(v interface{}) int {
 			buf.WriteString(fmt.Sprintf("%t-", v.(bool)))
 		}
 
+		if v, ok := m["automatic_upgrade_enabled"]; ok {
+			buf.WriteString(fmt.Sprintf("%t-", v.(bool)))
+		}
```

Comment:
this is a state migration of a given point in time - as such this needs to be reverted:

```suggestion
```